src/Pages/MembershipApplication/MembershipForm.js

```diff
@@ -38,50 +47,66 @@ export default function MembershipForm(props) {
       addon: !usernameAvailable && (
         <p className='unavailable'>User already exists!</p>
       ),
-      handleChange: e => setEmail(e.target.value)
+      ifLeftEmpty: !email && submitted && (
+        <p className='unavailable'>Email cannot be left empty</p>
+      ),
+      handleChange: e => setEmail(e.target.value),
+
     },
     {
       label: 'Password (8 or more characters)',
       type: 'password',
       addon: !passwordValid && (
         <p className='unavailable'>
-          Password requires one uppercase character and one number.
+          Password requires one uppercase character, one number
+          and at least 8 characters
+        </p>
+      ),
+      ifLeftEmpty: password.length<8 && submitted && (
+        <p className='unavailable'>
+          Password requires one uppercase character, one number
+          and at least 8 characters
         </p>
       ),
-      handleChange: e => setPassword(e.target.value)
+      handleChange: e => setPassword(e.target.value),
+
     }
   ];
 
+  function requiredFieldsEmpty() {
```

Comment:
Name of this function is misleading. `requiredFieldsEmpty` implies that the required fields have not been met. Rename to something like `requiredFieldsMet`